atom/browser/api/atom_api_app.cc

```diff
@@ -1162,6 +1195,30 @@ v8::Local<v8::Value> App::GetGPUFeatureStatus(v8::Isolate* isolate) {
   return mate::ConvertToV8(isolate, status ? *status : temp);
 }
 
+util::Promise* App::GetGPUInfo(const std::string& info_type) {
+  CHECK(!is_fetching_gpuinfo_);
```

Comment:
This is **scary**

Can't we just return `gpinfo_promise_` instead here 😄

Reply:
That's a good idea.